ddht/v5_1/alexandria/client.py

```diff
@@ -393,6 +397,37 @@ async def send_ack(
             node_id, endpoint, message, request_id=request_id
         )
 
+    async def send_locate(
+        self,
+        node_id: NodeID,
+        endpoint: Endpoint,
+        *,
+        content_key: ContentKey,
+        request_id: bytes,
+    ) -> bytes:
+        message = LocateMessage(LocatePayload(content_key))
+        return await self._send_request(
+            node_id, endpoint, message, request_id=request_id
+        )
+
+    async def send_locations(
+        self,
+        node_id: NodeID,
+        endpoint: Endpoint,
+        *,
+        advertisements: Sequence[Advertisement],
+        request_id: bytes,
+    ) -> int:
+        advertisement_batches = partition_advertisements(
+            advertisements, max_payload_size=MAX_PAYLOAD_SIZE,
+        )
+        num_batches = len(advertisement_batches)
+        for batch in advertisement_batches:
+            message = LocationsMessage(LocationsPayload(num_batches, batch))
```

Comment:
https://github.com/ethereum/ddht/blob/3719592a22f2d47754c34b6743f0c0de28ae1b16/ddht/v5_1/alexandria/payloads.py#L62-L64

https://github.com/ethereum/ddht/blob/d1b18da2f96dfb9eb6f849a599ff2cb53c195b21/tests/core/v5_1/alexandria/test_message_encoding.py#L112-L113

Not sure if this is worth changing, but `total` referring to the total # of batches rather than total number of advertisements was somewhat unclear to me.

Reply:
I'm sticking with the `total` for now, as it comes from the core discv5 spec terminology used in the base protocol and consumers of these APIs should never actually be dealing with the fields in these messages.